app/assets/javascripts/discourse/tests/acceptance/sticky-avatars-test.js

```diff
@@ -0,0 +1,35 @@
+import { acceptance, query } from "discourse/tests/helpers/qunit-helpers";
+import { test } from "qunit";
+import { settled, visit } from "@ember/test-helpers";
+import { later } from "@ember/runloop";
+
+acceptance("Sticky Avatars", function (needs) {
+  const container = document.getElementById("ember-testing-container");
+
+  needs.hooks.beforeEach(function () {
+    container.scrollTop = 0;
+  });
+
+  test("Adds sticky avatars when scrolling up", async function (assert) {
+    const appEvents = this.owner.lookup("service:app-events");
+
+    await visit("/t/internationalization-localization/280");
+
+    container.scrollTo(0, 800);
+    appEvents.trigger("topic:scrolled", 800);
+
+    container.scrollTo(0, 700);
+    appEvents.trigger("topic:scrolled", 700);
+
+    await settled();
+
+    const done = assert.async();
+    later(() => {
+      assert.ok(
+        query("#post_5").parentElement.classList.contains("sticky-avatar"),
+        "Sticky avatar is applied"
+      );
+      done();
+    }, 100);
```

Comment:
Doesn’t work without `later` and it wasn’t green on CI with 10 ms so we tried with 100 ms :grimacing: Seems to be working now.
At some point we should probably try to understand what’s the root cause (async stuff for sure :sweat_smile:)